intake/source/discovery.py

```diff
@@ -9,22 +9,253 @@
 import warnings
 import importlib
 import inspect
+import itertools
 import time
 import logging
 
+import entrypoints
+
 from .base import DataSource
 from ..catalog.base import Catalog
+from ..config import save_conf, conf, cfile
 logger = logging.getLogger('intake')
 
 
-def autodiscover(path=None, plugin_prefix='intake_'):
-    """Scan for Intake plugin packages and return a dict of plugins.
+def autodiscover(path=None, plugin_prefix='intake_', do_package_scan=True):
+    """Discover intake drivers.
+
+    In order of decreasing precedence:
+    - Respect the 'drivers' section of the intake configuration file.
+    - Find 'intake.drivers' entrypoints provided by any Python packages in the
+      environment.
+    - Search all packages in the environment for names that begin with
+      ``intake_``. Import them and scan them for subclasses of
+      ``intake.source.base.Plugin``. This was previously the *only* mechanism
+      for auto-discoverying intake drivers, and it is maintained for backward
+      compatibility. In a future release, intake will issue a warning if any
+      packages are located by the method that do not also have entrypoints.
+
+    Parameters
+    ----------
+    path : str or None
+        Default is ``sys.path``.
+    plugin_prefix : str
+        DEPRECATED. Default is 'intake_'.
+    do_package_scan : boolean
+        Default is True.
+
+    Returns
+    -------
+    drivers : dict
+        Name mapped to driver class.
+    """
+    # Discover drivers via package scan.
+    if do_package_scan:
+        package_scan_results = _package_scan(path, plugin_prefix)
+
+    # Discover drivers via entrypoints.
+    group = entrypoints.get_group_named('intake.drivers', path=path)
+    group_all = entrypoints.get_group_all('intake.drivers', path=path)
+    if group_all != group:
```

Comment:
group_all:list  , group:dict, so this will likely always be false ? – would need to check.
Force set-coercion, or compare length ?

Reply:
Ah, I meant this to be a len comparison.